### ISSUE TITLE
main callbacks return enum values?

### ISSUE THREAD

**Issue by slouken** [libsdl-org/SDL#10515]
SDL_AppEvent(), etc. all return int right now, and the recommended values are SDL_APP_CONTINUE, SDL_APP_FAILURE, etc.

Is this a free-form return value, or should we be returning an enum here?

**Comment by icculus** [comment]
It's an enum; you can't really return _anything_ you want here, as it doesn't become an exit code for the process.

The intention is that people that thought this was too wordy could use -1, 0, or 1, but as long as the enum values match those numbers, making it a formal enum is fine with me.
